pkg/store/bucket.go

```diff
@@ -3577,10 +3577,10 @@ func (r *bucketChunkReader) loadChunks(ctx context.Context, res []seriesEntry, a
 	bufPooled, err := r.block.chunkPool.Get(r.block.estimatedMaxChunkSize)
 	if err == nil {
 		buf = *bufPooled
+		defer r.block.chunkPool.Put(&buf)
 	} else {
 		buf = make([]byte, r.block.estimatedMaxChunkSize)
```

Comment:
Should we still put but back to the pool but extend buffer size to the bucket size?

Reply:
The buffer is only created when `ErrPoolExhausted` occurs, it should not be put back.